xtask/src/main.rs

```diff
@@ -202,6 +214,68 @@ fn main() -> Result<()> {
 // ----------------------------------------------------------------------------
 // Subcommands
 
+fn analyze_changelogs(workspace: &Path, args: AnalyzeChangelogArgs) -> Result<()> {
+    for package in args.packages {
+        let changelog_path = workspace.join(package.to_string()).join("CHANGELOG.md");
+        if !changelog_path.exists() {
+            continue;
+        }
+
+        let changelog = fs::read_to_string(changelog_path)?;
+        let changelog = parse_changelog::parse(&changelog)?;
+
+        let release = &changelog[0];
+        let release = markdown::to_mdast(&release.notes, &markdown::ParseOptions::default())
+            .expect("Unable to parse markdown");
+
+        let mut version_change: Option<Version> = None;
+        if let Some(children) = release.children() {
+            let mut i = 0;
+            loop {
+                if i >= children.len().saturating_sub(1) {
+                    break;
+                }
+
+                let Node::Heading(heading) = &children[i] else {
+                    bail!("Expected `Node::Heading(..), found something else");
+                };
+                let Node::Text(text) = &heading.children[0] else {
+                    bail!("Expected `Node::Text(..)`, found something else");
+                };
+
+                i += 1;
+
+                if let Node::List(list) = &children[i] {
+                    if !list.children.is_empty() {
+                        match text.value.to_lowercase().as_str() {
+                            // TODO: Once we move to `1.x.x` update to handle major versions
+                            "added" | "changed" | "removed" => {
+                                version_change = Some(Version::Minor);
```

Comment:
Adding things aren't always breaking changes.

Code quality could be improved a bit by deriving Ord on Version and using min/max

Reply:
Well I'm kind of assuming that for 1.0 the Minor here would be changed to Major and I'm preemptively saying that wouldn't be right 😅 If that's not the idea, just disregard me.

Reply:
I do generally agree that this won't always be perfect, but once we're at 1.0 we'll have more tooling (I guess at some point we'll wrangle semver checks) to help us